### PR TITLE
Webpack upgrade

### PR DESCRIPTION
**Pull request by runfalk**
This became a three part story. Firstly there was one more instance where `React.PropTypes` was used (it's still used in some dependencies).

The second commit is a complete update of Webpack to version 3.6.0. Since I can't test the electron build process yet I haven't done anything with that Webpack installation. Do we even want to use the current setup there? I took time to update Webpack, Babel and some of the loaders to get everything working. This fixes some parts of #449. One change I did was switching the source maps to proper ones instead of the `cheap-module-eval-source-map`. At least my Firefox versions (57 nightly and current stable) were unable to use those source maps. This adds about 10%-20% overhead to the JS build process. The maximum time on my (built 2010) computer is however still less than 10 seconds.

The third commit is a fix for broken usage of `this` in `multiWidgets.jsx`. I think this issue may be in master too, but I fixed it during my tests of the Webpack update to make sure nothing else was broken.

### CODE REVIEW

**Comment by runfalk** [comment]
I'm not sure why Travis fails on Node v8. That's what I've tested with locally. Could it be related to cached package versions? What's the stance on committing the `package-lock.json` to avoid works on my machine issues?

It seems that it ran on Node v8 on Appveyor with no issues.

File: lektor/admin/static/js/widgets/multiWidgets.jsx
Comment (nixjdm) [inline lektor/admin/static/js/widgets/multiWidgets.jsx:58]: This actually needs to stay using a normal function. Just changing it to an arrow function will result in the `this` inside the function being undefined. See https://github.com/lektor/lektor/pull/427 and the conversation on that pull request.

**Review by nixjdm — DISMISSED** [review]

**Comment by nixjdm** [comment]
Thanks @runfalk! It looks pretty good, but I want to take a little time to review it carefully and probably test it out manually before merging, so please be a little patient. I don't have a well developed opinion on the package-lock file, so I'll do some reading. The only bad part about it that I'm aware of is excess commits / noise it might cause, which probably isn't a huge deal. React isn't my forte. On that note, @vanadium23 or @singingwolfboy I'd appreciate another set of eyes as well.

File: lektor/admin/static/js/widgets/multiWidgets.jsx
Comment (runfalk) [reply lektor/admin/static/js/widgets/multiWidgets.jsx:58]: Before I changed this to an arrow function I got errors about `this` being undefined. After changing it to an arrow function, my model worked as intended. Are you certain this particular case needs to be a normal anonymous function?

This was tested in firefox stable and nightly.

**Comment by runfalk** [comment]
If necessary I can put the commits into separate pull requests, but since both commit 1 and 3 are so small I figured I'd just piggy back them into this one.

File: lektor/admin/static/js/widgets/multiWidgets.jsx
Comment (runfalk) [reply lektor/admin/static/js/widgets/multiWidgets.jsx:58]: I did some more reading on [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) and the important thing here is that they'll not bind `this`, meaning they are free to use the outer scope's `this`. Since both `this.isActive` and `this.onChange` is part of the outer scope this should/must be an arrow function.

Comment (nixjdm) [reply lektor/admin/static/js/widgets/multiWidgets.jsx:58]: It seems I 'fixed' a false positive in #427. This function can be an arrow function (I'll remove the change request). As an arrow function it works fine for me, which you're right should be expected since there is a functional this in the outer scope. Though, I'm not getting an additional 'undefined' in app.js as a normal function (as it was even in the initial commit), and I would expect that to still work too. I'm curious about that. Where are you seeing the error?

**Comment by nixjdm** [comment]
@runfalk You don't have to split the commits up. Thanks for the offer though.

File: lektor/admin/static/js/widgets/multiWidgets.jsx
Comment (runfalk) [reply lektor/admin/static/js/widgets/multiWidgets.jsx:58]: I haven't isolated it exactly, but it might be related to a list of checkboxes. For this particular case I've made a plugin that reads categories from a databag and displays the contents as checkboxes, but I assume any collection of checkboxes will trigger it.

**Comment by nixjdm** [comment]
This fixes #454. I'm almost done reviewing. Hopefully we can pull this soon.

**Comment by nixjdm** [comment]
@runfalk Thank you for your work and your patience! I had to go slow to keep looking things up and reading, but it makes sense and looks like some good cleanup, and it works. lgtm!

**Comment by nixjdm** [comment]
@runfalk And by the way, don't worry about the JS environment of the desktop app right now. It's got it's own headaches https://github.com/lektor/lektor/issues/420

**Comment by runfalk** [comment]
Thank you for merging this!

**Pull request merged**
